### PR TITLE
Export duplicates fix

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

Fixes bug where Search lib would slice filtered reports view results with negative end index when API export would paginate. 
Fixes bug where API export would not convert "true" and "false" filters to Boolean. 
Enables sharedLibs unit tests in Travis, but disables bulk-doc-utils unit tests because they're broken. 

medic/medic-webapp#4610

# Review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- [ ] Documented: Announced in Changes.md in plain English. Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

### CODE REVIEW

File: shared-libs/search/src/search.js [hunk 1]
```diff
@@ -8,6 +8,10 @@ var _ = require('underscore'),
 module.exports = function(Promise, DB) {
   // Get the subset of rows, in appropriate order, according to options.
   var getPageRows = function(type, rows, options) {
+    if (rows.length < options.skip) {
+      return [];
```
Comment (SCdF) [inline shared-libs/search/src/search.js]: I don't get how this works, for me it needs an explanatory comment.

File: Gruntfile.js [hunk 2]
```diff
@@ -276,15 +276,15 @@ module.exports = function(grunt) {
       sharedLibUnit: {
         cmd:  function() {
           var sharedLibs = [
-            'bulk-docs-utils',
+            //'bulk-docs-utils',
```
[the remark this answers is not in the document]
Comment (dianabarsan) [reply Gruntfile.js]: Exactly.

**Comment by dianabarsan** [comment]
I'm not exactly sure what to do with bulk docs utils. The change that causes the tests to fail was not forward ported to 3.0.0 - in 3.0.0 bulk docs utils tests are run in Travis and do not fail. I was planning to raise a separate issue for them.

**Comment by dianabarsan** [comment]
I had an attempt to fix `bulk-docs-utils` tests (and dependencies - npm issues)  in this PR, but then remembered that the commit that causes the tests to fail (https://github.com/medic/medic-webapp/commit/4310f14e416ed4da8cff18f10d4ce87ebc3cc27d) was not forward ported to 3.0.0.

If this needs to be forward ported to 3.0.0, it should at least not break it :)

File: shared-libs/search/src/search.js (hunk 1 above)
Comment (dianabarsan) [reply shared-libs/search/src/search.js]: Do you think the comment I added is clear enough?

**Pull request merged**